common/testing.cpp

```diff
@@ -55,6 +55,12 @@ std::string TemporaryDirectory::Path() const {
 	return path_;
 }
 
+void TemporaryDirectory::CreateSubDirectory(const string &dirname) {
+	fs::path sub_path {path_};
+	sub_path.append(dirname);
+	fs::create_directory(sub_path);
```

Comment:
Handy!

I would `EXPECT_TRUE` around this though.

Reply:
Yeh, I considered it - however, it is not done during the initialiser either. 

Should i add it to both?

Reply:
Please do! It won't cause the test to fail immediately (even `ASSERT` just returns, ideally it should `throw` its way out), but at least you can see in the log what happened before other mysterious failures.

Reply:
https://github.com/mendersoftware/mender/pull/1353/commits/49afd4d292a38834d8c6e60e9b2d0daf8ab96c34

Reply:
I did throw, but re-reading this I think I might have misunderstood you. You do mean `ASSERT` is the way to go?

Reply:
Ah yes, sorry if that was confusing. I meant that the Google test framework could have invented something smarter, such as `throw`, or even `longjmp`, to get out of deep asserts, but they only return. I think we should keep our own usage of `throw` to a minimum, but it's not completely out of the question. For example we do throw [in the `TestEventLoop`](https://github.com/mendersoftware/mender/blob/e3f189f2b87e79893ae5902f93a81bc5e22cd57f/common/testing.hpp#L74), because not doing so would frequently hang the test instead, which kind of defeats the purpose.

You can choose if you want to use it, but failure in this location is probably quite rare, so I don't think it's important to use `throw` here.

Reply:
Changed it to `ASSERT`

Reply:
Actually, on second try, we do need to throw, just because of what you said before, since this is in the constructor in the one instance.

The other one I'm keeping `Assert` :)

Reply:
Sure, then it's justified.